From 9a539c17d419b5b49a7afefff4d15cc5076461d9 Mon Sep 17 00:00:00 2001
From: Judith Mendez <jm@ti.com>
Date: Mon, 20 Oct 2025 12:59:47 -0500
Subject: [PATCH] TI: arm64: dts: ti: k3-am62: Move wakeup-source

Move wakeup-source to overlays. This fixes deep sleep wakeup with
rtcwake. Additionally, wakeup source configuration is not complete without
the additional pinmux configuration anyways.

Fixes: 2733af6d381d ("FROMLIST: arm64: dts: ti: k3-am62: Mark mcu_mcan0/1 as wakeup-source")
Fixes: e6aced8fa0f1 ("PENDING: can: m_can: Enable pinctrl wakeup without WoL enabled")
Signed-off-by: Judith Mendez <jm@ti.com>
Tested-by: Kendall Willis <k-willis@ti.com>
Reviewed-by: Kendall Willis <k-willis@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62-mcu.dtsi                         | 2 --
 arch/arm64/boot/dts/ti/k3-am62a-mcu.dtsi                        | 2 --
 arch/arm64/boot/dts/ti/k3-am62p-j722s-common-mcu.dtsi           | 2 --
 arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-io-ddr-wkup-sources.dtso | 2 ++
 arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso        | 2 ++
 5 files changed, 4 insertions(+), 6 deletions(-)

Upstream-Status: Backport

diff --git a/arch/arm64/boot/dts/ti/k3-am62-mcu.dtsi b/arch/arm64/boot/dts/ti/k3-am62-mcu.dtsi
index cc693be47df08..68e906796aefe 100644
--- a/arch/arm64/boot/dts/ti/k3-am62-mcu.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62-mcu.dtsi
@@ -160,7 +160,6 @@
 		clocks = <&k3_clks 188 6>, <&k3_clks 188 1>;
 		clock-names = "hclk", "cclk";
 		bosch,mram-cfg = <0x0 128 64 64 64 64 32 32>;
-		wakeup-source = "suspend", "poweroff";
 		status = "disabled";
 	};
 
@@ -173,7 +172,6 @@
 		clocks = <&k3_clks 189 6>, <&k3_clks 189 1>;
 		clock-names = "hclk", "cclk";
 		bosch,mram-cfg = <0x0 128 64 64 64 64 32 32>;
-		wakeup-source = "suspend", "poweroff";
 		status = "disabled";
 	};
 
diff --git a/arch/arm64/boot/dts/ti/k3-am62a-mcu.dtsi b/arch/arm64/boot/dts/ti/k3-am62a-mcu.dtsi
index ef4e07f84e1a1..3df07602df7b3 100644
--- a/arch/arm64/boot/dts/ti/k3-am62a-mcu.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62a-mcu.dtsi
@@ -173,7 +173,6 @@
 		clocks = <&k3_clks 188 6>, <&k3_clks 188 1>;
 		clock-names = "hclk", "cclk";
 		bosch,mram-cfg = <0x0 128 64 64 64 64 32 32>;
-		wakeup-source = "suspend", "poweroff";
 		status = "disabled";
 	};
 
@@ -186,7 +185,6 @@
 		clocks = <&k3_clks 189 6>, <&k3_clks 189 1>;
 		clock-names = "hclk", "cclk";
 		bosch,mram-cfg = <0x0 128 64 64 64 64 32 32>;
-		wakeup-source = "suspend", "poweroff";
 		status = "disabled";
 	};
 
diff --git a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-mcu.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-mcu.dtsi
index f96e9f9dae813..0dbb5b4e09d10 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-mcu.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-mcu.dtsi
@@ -177,7 +177,6 @@
 		interrupts = <GIC_SPI 26 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 27 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "int0", "int1";
-		wakeup-source = "suspend", "poweroff";
 		status = "disabled";
 	};
 
@@ -193,7 +192,6 @@
 		interrupts = <GIC_SPI 28 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 29 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "int0", "int1";
-		wakeup-source = "suspend", "poweroff";
 		status = "disabled";
 	};
 
diff --git a/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-io-ddr-wkup-sources.dtso b/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-io-ddr-wkup-sources.dtso
index a507b99a61c8c..02f4ab9903c2c 100644
--- a/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-io-ddr-wkup-sources.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-io-ddr-wkup-sources.dtso
@@ -59,6 +59,7 @@
 	pinctrl-names = "default", "wakeup";
 	pinctrl-0 = <&mcu_mcan0_tx_pins_default>, <&mcu_mcan0_rx_pins_default>;
 	pinctrl-1 = <&mcu_mcan0_tx_pins_default>, <&mcu_mcan0_rx_pins_wakeup>;
+        wakeup-source = "suspend", "poweroff";
 	status = "okay";
 };
 
@@ -66,5 +67,6 @@
 	pinctrl-names = "default", "wakeup";
 	pinctrl-0 = <&mcu_mcan1_tx_pins_default>, <&mcu_mcan1_rx_pins_default>;
 	pinctrl-1 = <&mcu_mcan1_tx_pins_default>, <&mcu_mcan1_rx_pins_wakeup>;
+        wakeup-source = "suspend", "poweroff";
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso b/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso
index 045ab6c5ed3d1..7e23950565a07 100644
--- a/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso
@@ -112,6 +112,7 @@
 	pinctrl-names = "default", "wakeup";
 	pinctrl-0 = <&mcu_mcan0_tx_pins_default>, <&mcu_mcan0_rx_pins_default>;
 	pinctrl-1 = <&mcu_mcan0_tx_pins_default>, <&mcu_mcan0_rx_pins_wakeup>;
+	wakeup-source = "suspend", "poweroff";
 	status = "okay";
 };
 
@@ -119,5 +120,6 @@
 	pinctrl-names = "default", "wakeup";
 	pinctrl-0 = <&mcu_mcan1_tx_pins_default>, <&mcu_mcan1_rx_pins_default>;
 	pinctrl-1 = <&mcu_mcan1_tx_pins_default>, <&mcu_mcan1_rx_pins_wakeup>;
+	wakeup-source = "suspend", "poweroff";
 	status = "okay";
 };
